### PR TITLE
v1: use nested `.v1` layer name instead of `-v1` suffix

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Follow-up to #1412 and #1415. Renamed the layer and also specified layer order upfront (in `global.css`) to ensure correct order is used.

## Testing

N/A

## Docs

updated changelog.

### CODE REVIEW

File: packages/itwinui-css/src/style/global.scss [hunk 1]
```diff
@@ -3,7 +3,9 @@
 @import './index';
 @import './anchor';
 
-@layer itwinui-v1 {
+@layer itwinui-v1, itwinui.v1, itwinui.v2, itwinui.v3;
```
Comment (adamhock) [inline packages/itwinui-css/src/style/global.scss:6]: Can the `itwinui-v1` layer be removed?

**Review by adamhock — COMMENTED** [review]
The two visual tests that failed in the CI also failed when building locally

File: packages/itwinui-css/src/style/global.scss (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-css/src/style/global.scss:6]: no, we need both. https://github.com/iTwin/iTwinUI/pull/1412#discussion_r1263716496

users might have the old stylesheet included somewhere

**Comment by mayank99** [comment]
> The two visual tests that failed in the CI also failed when building locally

unrelated to this PR and it looks fine, so i'm not worried about the 0.5% difference. this is the old v0 css codebase that will likely never be updated again.

**Review by adamhock — APPROVED** [review]

**Review by gretanausedaite — APPROVED** [review]

**Pull request merged**